packages/core/src/sagas/startSpeakActivityOnPostActivitySaga.js

```diff
@@ -0,0 +1,25 @@
+import {
+  put,
+  takeEvery
+} from 'redux-saga/effects';
+
+import whileConnected from './effects/whileConnected';
+
+import { POST_ACTIVITY_PENDING } from '../actions/postActivity';
+import startSpeakingActivity from '../actions/startSpeakingActivity';
+import stopSpeakingActivity from '../actions/stopSpeakingActivity';
+
+export default function* () {
+  yield whileConnected(function* () {
+    yield takeEvery(
+      ({ meta, payload, type }) => (
```

Comment:
inlining this takes away from the readability for me. can it be defined outside the function and given a good name?

Reply:
Refactored out into a separate function.